packages/front-end/components/Metrics/MetricForm/index.tsx

```diff
@@ -427,6 +444,37 @@ const MetricForm: FC<MetricFormProps> = ({
     }
   }, [type, form]);
 
+  const tableOptions: { label: string; value: string }[] = [];
+
+  const { data: TableData } = useApi<{
+    informationSchema: InformationSchemaInterface;
+  }>(`/datasource/${selectedDataSource.id}/schema`);
+
+  if (TableData?.informationSchema?.databases.length) {
+    TableData.informationSchema.databases.forEach((database) => {
+      database.schemas.forEach((schema) => {
+        schema.tables.forEach((table) => {
```

Comment:
Should these 3 lines have optional chaining? I think there was a bug earlier related to this. Schemas was undefined in a database or something like that.